Copy-file cannot copy directory

Gradle Profiler version 0.21.0

Documentation states that copy-file: Copies a file or a directory from one location to another.
But, specifying a directory results in an Exception. 

```
copy-file = {
        source = "foo-reference"
        target = "foo"
        schedule = BUILD
    }
```

```
Caused by: java.io.FileNotFoundException: /Users/adecker/src/android/foo-reference (Is a directory)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:213)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:152)
        at com.google.common.io.Files$FileByteSource.openStream(Files.java:134)
        at com.google.common.io.Files$FileByteSource.openStream(Files.java:122)
        at com.google.common.io.ByteSource.copyTo(ByteSource.java:278)
        at com.google.common.io.Files.copy(Files.java:328)
        at org.gradle.profiler.mutations.CopyFileMutator.executeOnSchedule(CopyFileMutator.java:33)
        ... 9 more
```

--- a/src/main/java/org/gradle/profiler/mutations/CopyFileMutator.java
+++ b/src/main/java/org/gradle/profiler/mutations/CopyFileMutator.java
@@ -3,6 +3,8 @@ package org.gradle.profiler.mutations;
 import com.google.common.base.Strings;
 import com.google.common.io.Files;
 import com.typesafe.config.Config;
+
+import org.apache.commons.io.FileUtils;
 import org.gradle.profiler.BuildMutator;
 import org.gradle.profiler.CompositeBuildMutator;
 import org.gradle.profiler.ConfigUtil;
@@ -30,7 +32,11 @@ public class CopyFileMutator extends AbstractFileSystemMutator {
             if (!target.exists()) {
                 Files.createParentDirs(target);
             }
-            Files.copy(source, target);
+            if(source.isDirectory()) {
+                FileUtils.copyDirectory(source, target);
+            } else {
+                Files.copy(source, target);
+            }
         } catch (IOException e) {
             throw new UncheckedIOException("Failed to copy '" + source.getAbsolutePath() + "' to '" + target.getAbsolutePath() + "'", e);
         }

--- a/src/test/groovy/org/gradle/profiler/mutations/CopyFileMutatorTest.groovy
+++ b/src/test/groovy/org/gradle/profiler/mutations/CopyFileMutatorTest.groovy
@@ -27,6 +27,33 @@ class CopyFileMutatorTest extends AbstractMutatorTest {
         target.text == expectedContents
     }
 
+    def "copies directory and contents from source to target"() {
+        def testDir = tmpDir.newFolder()
+
+        def expectedContents = "Copy file from source to target"
+        def source = new File(testDir, "source/file.txt")
+        source.parentFile.mkdirs()
+        source.text = expectedContents
+
+        def target = new File(testDir, "nested/target/file.txt")
+
+        def spec = mockConfigSpec("""{
+            copy-file = {
+                source = "source"
+                target = "nested/target"
+            }
+        }""")
+        _ * spec.projectDir >> testDir
+
+        when:
+        def mutator = new CopyFileMutator.Configurator().configure("copy-file", spec)
+        mutator.beforeScenario(scenarioContext)
+
+        then:
+        target.exists()
+        target.text == expectedContents
+    }
+
     def "copies multiple sets of source and target"() {
         def testDir = tmpDir.newFolder()
 

